src/logics/SubstLoopBreaker.cpp

```diff
@@ -248,7 +250,7 @@ Map<PTRef,PtAsgn,PTRefHash> SubstLoopBreaker::operator() (const vec<Map<PTRef,Pt
         uniq(startNodes);
 //        printGraphAndLoops(startNodes, loops);
     }
-    return constructLooplessSubstitution(std::move(substs));
+    return constructLooplessSubstitution(std::move(substs)); // Why don't we need the move contructor here, given that Map is a template?
```

Comment:
Why do you think being or not being a template makes a difference?

Reply:
I forgot that comment there, I'll remove it.  I actually later figured out that when map comes from that function it already has been cast to the movable type by std::move.

But I was expecting the copy elusion to work in that function (which it does not) without std::move.  I figured out that for copy elusion there needs to be a non-template constructor, and I'm now thinking that that's the reason.